src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1398,6 +1435,75 @@ private MqttServerStream resolvePublisher(
             return stream;
         }
 
+        private void decodeSession(
+            long traceId,
+            long authorization,
+            int flags)
+        {
+            final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
+            final int topicKey = topicKey(topic);
+
+            final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
+
+            MqttServerStream stream;
+            if (route != null)
+            {
+                final long resolvedId = route.correlationId();
+
+                final JsonObject sessionObj = OBJECT_BUILDER.add(SESSION_EXPIRES_AT_NAME, sessionExpiresAt)
+                                                            .add(WILL_DELAY_NAME, mqttConnectPayloadRO.willDelay)
+                                                            .build();
+
+                final OctetsFW payload = sessionPayloadRW.wrap(payloadBuffer, 0, payloadBuffer.capacity())
+                                                         .set(sessionObj.toString().getBytes(UTF_8))
+                                                         .build();
+
+                final MqttDataExFW dataEx = mqttDataExRW.wrap(dataExtBuffer, 0, dataExtBuffer.capacity())
+                                                        .typeId(mqttTypeId)
+                                                        .topic(String.format(SESSION_TOPIC_FORMAT, clientId))
+                                                        .build();
+
+                stream = streams.computeIfAbsent(topicKey, s -> new MqttServerStream(resolvedId, 0, topic));
+
+                final int reserved = payload.sizeof() + mqttConnectPayloadRO.willPayload.sizeof() + stream.initialPadding;
+
+                stream.doApplicationBeginOrFlush(traceId, authorization, affinity, topic, NO_FLAGS, 0, PUBLISH_AND_SUBSCRIBE);
+                stream.doApplicationData(traceId, authorization, reserved, payload, dataEx);
+
+                if ((flags & WILL_FLAG_MASK) != 0)
+                {
+                    int willDataFlags = 0;
+
+                    if (mqttConnectPayloadRO.willRetain != 0)
+                    {
+                        willDataFlags = willDataFlags & mqttConnectPayloadRO.willRetain;
+                    }
+
+                    if (mqttConnectPayloadRO.willQos != 0)
+                    {
+                        willDataFlags = willDataFlags & mqttConnectPayloadRO.willQos;
+                    }
```

Comment:
Pass in whatever is needed by the method rather than referring to global shared `mqttConnectPayloadRO` state.